src/bot_events.py

```diff
@@ -0,0 +1,92 @@
+from types import SimpleNamespace
+import inspect
+
+_callbacks = { 
+    # even though the hook.Add function will automatically key:value and put it in here if it doesn't exist,
+    # be nice to other developers and make sure you put it in here so they know it exists
+    # thanks
+    
+    # bot hooks
+    "on_bot_init": {},
+
+    "on_create_game": {},
+    "on_end_game": {},
+
+    "on_run_pindle": {},
```

Comment:
Consider that the interface for bots effectively allow you to implement a singular callback function for each of these "on_run_foo"s at the moment.  This changeset is basically just letting you define multiple callbacks for when you enter a pindle run.  Is there actually a reason you would ever want to do that? In addition, when you register an `on_run_pindle` callback, it's may be less clear what arguments your callback function will receive than it previously was when you are implementing an abstract method.